deps/build.jl

```diff
@@ -15,36 +20,28 @@ const REGIONS = (
 isdir(TZDATA_DIR) || mkdir(TZDATA_DIR)
 isdir(COMPILED_DIR) || mkdir(COMPILED_DIR)
 
-# TODO: Downloading from the IANA source fails regularly. We should attempt to find
-# alternative sources.
 info("Downloading TZ data")
-@sync for region in REGIONS
-    @async begin
-        remote_file = "ftp://ftp.iana.org/tz/data/" * region
-        region_file = joinpath(TZDATA_DIR, region)
-        remaining = 3
-
-        while remaining > 0
-            try
-                # Note the destination file will be overwritten upon success.
-                download(remote_file, region_file)
-                remaining = 0
-            catch e
-                if isa(e, ErrorException)
-                    if remaining > 0
-                        remaining -= 1
-                    elseif isfile(region_file)
-                        warn("Falling back to old region file $region. Unable to download: $remote_file")
-                    else
-                        error("Missing region file $region. Unable to download: $remote_file")
-                    end
-                else
-                    rethrow()
-                end
-            end
-        end
+archive = ""
+for url in URLS
+    try
+        archive = download(url)
+        break
+    catch
+        warn("Failed to download TZ data from: $url")
     end
 end
+isfile(archive) || error("Unable to download TZ data")
+
+info("Extracting TZ data")
+@unix_only function extract(archive, directory, files)
+    run(`tar xvf $archive --directory=$directory $files`)
+end
+@windows_only function extract(archive, directory, files)
+    run(pipeline(`7z x $archive -y -so`, `7z x -si -y -ttar -o$directory $files`))
+end
+
+extract(archive, TZDATA_DIR, REGIONS)
+rm(archive)
```

Comment:
We don't have to remove the archive. The file is being downloaded to a temporary file so this is just cleanup. You think it would be useful for debugging to just leave the archive?